ci-operator/step-registry/acm/mch/acm-mch-commands.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+set -x
+set -o nounset
+set -o errexit
+set -o pipefail
+
+echo "This is the acm-mch-command.sh space running"
+
+export KUBECONFIG=${SHARED_DIR}/kubeconfig
+
+echo "Apply multiclusterhub"
+# apply MultiClusterHub crd
+oc apply -f - <<EOF
+apiVersion: operator.open-cluster-management.io/v1
+kind: MultiClusterHub
+metadata:
+  name: multiclusterhub
+  namespace: ocm
+spec: {}
+EOF
+
+# can't wait before the resource exists. Need to sleep a bit before start watching
+sleep 60
+
+RETRIES=30
+for i in $(seq "${RETRIES}"); do
+  if [[ $(oc get mch -n ocm -o=jsonpath='{.items[0].status.phase}') == "Running" ]]; then
+    echo "MCH is Running"
+    break
+  else
```

Comment:
You may want to add a check in this else statement to see if you are on the last retry, if you are exit the script with an error.

Here is what I did with MTR:

```bash
RETRIES=30
for try in $(seq "$RETRIES"); do
    DEPLOYMENT_EXISTS=$(oc get windup -n mtr -o=jsonpath='{.items[0].status.conditions[0].status}')
    READY=$(oc get windup -n mtr -o=jsonpath='{.items[0].status.conditions[1].status}')

    if [ DEPLOYMENT_EXISTS == "True" ] && [ READY == "True"]; then
        echo "Windup deployment exists and is ready."
        break
    else
        if [ $try == $RETRIES ]; then
            echo "Error deploying Windup, exiting now"
            exit 1
        fi
        echo "Try ${try}/${RETRIES}: Windup deployment is not ready. Checking again in 30 seconds"
        sleep 30
    fi
done
```